docs/en/sql-reference/statements/update.md

```diff
@@ -0,0 +1,115 @@
+   - Lightweight updates are designed to update small amounts of rows (up to about 10% of the table). If you need to update a larger amount, it is recommended to use the [ALTER TABLE ... UPDATE](/sql-reference/statements/alter/update) mutation
+
+## Concurrent operations {#concurrent-operations}
+
+Lightweight updates don't wait for currently running merges/mutations to complete unlike heavy mutations.
+The consistency of concurrent lightweight updates is controlled by settings [update_sequential_consistency](/operations/settings/settings#update_sequential_consistency) and [update_parallel_mode](/operations/settings/settings#update_parallel_mode).
+
+## Update permissions {#update-permissions}
+
+`UPDATE` requires the `ALTER UPDATE` privilege. To enable `UPDATE` statements on a specific table for a given user, run:
+
+```sql
+GRANT ALTER UPDATE ON db.table TO username;
+```
+
+## Details of the implementation {#details-of-the-implementation}
+
+Patch parts are the same as the regular parts, but contain only updated columns and several system columns:
+  - `_part` - the name of the original part
+  - `_part_offset` - the row number in the original part
+  - `_block_number` - the block number of the row in the original part
+  - `_block_offset` - the block offset of the row in the original part
+  - `_data_version` - the data version of the updated data (block number allocated for the UPDATE query)
+
+On average it gives about 40 bytes (uncompressed data) of overhead per updated row in the patch parts.
+System columns help to find rows in original part which should be updated.
+System columns are related to the [virtual columns](/engines/table-engines/mergetree-family/mergetree.md/#virtual-columns) in the original part, which are added for reading if patch parts should be applied.
+Patch parts are sorted by `_part` and `_part_offset`.
+
+Patch parts belong to the different partitions than the original part.
+The partition id of the patch part is `patch-<hash of column names in patch part>-<original_partition_id>`.
+Therefore patch parts with different columns are stored in different partitions.
+For example three updates `SET x = 1 WHERE <cond>`, `SET y = 1 WHERE <cond>` and `SET x = 1, y = 1 WHERE <cond>` will create three patch parts in three different partitions.
+
+Patch parts can be merged among themselves to reduce the amount of applied patches on `SELECT` queries and reduce the overhead. Merge of patch parts uses [Replacing](/engines/table-engines/mergetree-family/replacingmergetree) merge algorithm with `_data_version` as a version column.
+Therefore patch part always stores the latest version for each updated row in part.
+
+Lightweight updates don't wait for currently running merges and mutations to finish and always use current snapshot of data parts to execute an update and produce a patch part.
+Because of that there can be two cases of applying patch parts.
```

Comment:
```suggestion
Lightweight updates don't wait for currently running merges and mutations to finish and always use a current snapshot of data parts to execute an update and produce a patch part.
Because of that there can be two cases of applying patch parts.
```